packages/wrangler/src/publish.ts

```diff
@@ -164,14 +157,45 @@ export default async function publish(props: Props): Promise<void> {
     let migrations;
     if (config.migrations.length > 0) {
       // get current migration tag
-      const scripts = await fetchResult<
-        { id: string; migration_tag: string }[]
-      >(`/accounts/${accountId}/workers/scripts`);
-      const script = scripts.find(({ id }) => id === scriptName);
+      type ScriptData = { id: string; migration_tag?: string };
+      let script: ScriptData | undefined;
+      if (!props.legacyEnv) {
+        try {
+          if (props.env) {
+            const scriptData = await fetchResult<{
+              script: ScriptData;
+            }>(
+              `/accounts/${accountId}/workers/services/${scriptName}/environments/${props.env}`
+            );
+            script = scriptData.script;
+          } else {
+            const scriptData = await fetchResult<{
+              default_environment: {
+                script: ScriptData;
+              };
+            }>(`/accounts/${accountId}/workers/services/${scriptName}`);
+            script = scriptData.default_environment.script;
+          }
+        } catch (err) {
+          if (
+            [10090, 10092].includes((err as { code: number }).code) === false
```

Comment:
i feel like this should just be `![10090, 10092].includes((err as { code: number }).code)`, right?

Reply:
Can we also add comments for what these error numbers mean

Reply:
heh I'd originally had it in that form, happy to revert to it. I'll add comments for the codes.